Include/internal/pycore_bitutils.h

```diff
@@ -82,6 +84,45 @@ _Py_bswap64(uint64_t word)
 }
 
 
+// Population count: count the number of 1's in 'u'
+// (number of bits set to 1), also known as the hamming weight.
+//
+// Implementation note. CPUID is not used, to test if x86 POPCNT instruction
+// can be used, to keep the implementation simple. For example, Visual Studio
+// __popcnt() is not used this reason. The clang and GCC builtin function can
+// use the x86 POPCNT instruction if the target architecture has SSE4a or
+// newer.
+static inline int
+_Py_popcount32(uint32_t x)
+{
+#if (defined(__clang__) || defined(__GNUC__)) && SIZEOF_INT == 4
+    return __builtin_popcount(x);
+#elif (defined(__clang__) || defined(__GNUC__)) && SIZEOF_INT == 8
+    return __builtin_popcountl(x);
+#else
+    // 32-bit SWAR (SIMD Within A Register) popcount
+
+    // Binary: 0 1 0 1 ...
+    const uint32_t M1 = UINT32_C(0x55555555);
+    // Binary: 00 11 00 11. ..
+    const uint32_t M2 = UINT32_C(0x33333333);
+    // Binary: 0000 1111 0000 1111 ...
+    const uint32_t M4 = UINT32_C(0x0F0F0F0F);
+    // 256^4 + 256^3 + 256^2 + 256^1
```

Comment:
The exponents need fixing in this comment: this is actually `256**3 + 256**2 + 256**1 + 256**0`. (I'd spell it with a `**` rather than a `^`, since `^` is bitwise xor in both C and Python.)

Reply:
Yeah, I hesitate to use **. Comment updated to use **.